### PR TITLE
feat: models

### PR DESCRIPTION
**Pull request by macintoshpie**
Adds ability to export 3 and 4 parameter models into bsync.
5P is not handled b/c it is failing to complete with our SEED test data currently.

### CODE REVIEW

File: R/bsync_utils.R [hunk 1]
```diff
@@ -301,9 +295,13 @@ bs_gen_dm_nmecr <- function(nmecr_baseline_model, x,
     "Hourly" = "Hour",
     "SLR" = "2 parameter simple linear regression",
     "Three Parameter Heating" = "3 parameter heating change point model",
```
Comment (corymosiman12) [inline R/bsync_utils.R:297]: @macintoshpie should we remove the redundant ones?  i.e. I assume the keys with 'Three Parameter Heating' and '3PH' are the same, and you are probably only using '3PH'...unless I missed something.

**Review by corymosiman12 — APPROVED** [review]
Minor changes - seems simple enough.

File: R/bsync_utils.R (hunk 1 above)
Comment (macintoshpie) [reply R/bsync_utils.R:297]: We need both from what I can tell b/c it depends on what you pass to nmecr (which allows both)
```r
model <- nmecr::model_with_CP(bsyncr_df,
                                nmecr::assign_model_inputs(regression_type = "Three Parameter Cooling"))
print(model$model_input_options$regression_type)
# prints "Three Parameter Cooling"
model <- nmecr::model_with_CP(bsyncr_df,
                                nmecr::assign_model_inputs(regression_type = "3PC"))
print(model$model_input_options$regression_type)
# prints "3PC"
```

**Review by corymosiman12 — APPROVED** [review]
Looks good.  Anyway to add some testing?  Or just add an issue to begin define testing? I think we can probably just have a predefined:
- Input file
- Output file (expected)

And then the test just compares a new output file to the expected output file.  It can be more 'outcome based' than unit test based - make sense?

**Comment by macintoshpie** [comment]
yeah we should definitely add some testing at some point

**Pull request merged**